orchestrator/graphql/loaders/subscriptions.py

```diff
@@ -1,224 +1,37 @@
-        )
-        .join(ResourceTypeTable)
-        .filter(ResourceTypeTable.resource_type.in_(RELATION_RESOURCE_TYPES))
-        .filter(in_use_by_instances.subscription_id.in_(unique_subscription_ids))
-        .filter(depends_on_subscriptions.status.in_(filter_statuses))
-    )
-
-    return _get_resource_type_relations(query_get_depends_on_ids)
-
-
-async def _get_in_use_by_relations(subscription_ids: list[UUID], filter_statuses: list[str]) -> list[Relation]:
-    if RELATION_RESOURCE_TYPES:
-        # Find relations through resource types
-        resource_type_relations = await _get_in_use_by_resource_type_relations(subscription_ids, filter_statuses)
-    else:
-        resource_type_relations = []
-    # Find relations through instance hierarchy
-    instance_relations = await _get_in_use_by_instance_relations(subscription_ids, filter_statuses)
-    return list(chain(resource_type_relations, instance_relations))
-
-
-async def _get_depends_on_relations(subscription_ids: list[UUID], filter_statuses: list[str]) -> list[Relation]:
-    if RELATION_RESOURCE_TYPES:
-        # Find relations through resource types
-        resource_type_relations = await _get_depends_on_resource_type_relations(subscription_ids, filter_statuses)
-    else:
-        resource_type_relations = []
-    # Find relations through instance hierarchy
-    instance_relations = await _get_depends_on_instance_relations(subscription_ids, filter_statuses)
-    return list(chain(resource_type_relations, instance_relations))
-
-
-async def in_use_by_subs_loader(keys: list[tuple[UUID, list[str] | None]]) -> list[list[SubscriptionTable]]:
+async def in_use_by_subs_loader(keys: list[tuple[UUID, tuple[str, ...]]]) -> list[list[SubscriptionTable]]:
     """GraphQL dataloader to efficiently get the in_use_by SubscriptionTables for multiple subscription_ids."""
     subscription_ids = [key[0] for key in keys]
-    filter_statuses: list[str] = keys[0][1] or SubscriptionLifecycle.values()
-
-    in_use_by_relations = await _get_in_use_by_relations(subscription_ids, filter_statuses)
-
-    # Retrieve SubscriptionTable for all unique inuseby ids
-    unique_in_use_by_ids = {row.in_use_by_sub_id for row in in_use_by_relations}
-    _in_use_by_subs = db.session.execute(
-        select(SubscriptionTable).filter(SubscriptionTable.subscription_id.in_(unique_in_use_by_ids))
-    ).scalars()
-    in_use_by_subs = {subscription.subscription_id: subscription for subscription in _in_use_by_subs}
+    filter_statuses: tuple[str, ...] = keys[0][1] or tuple(SubscriptionLifecycle.values())
 
-    # group (more_itertools.bucket doesn't seem to work for tuple of uuids)
-    subscription_in_use_by_ids: dict[UUID, list[UUID]] = {}
-    for relation in in_use_by_relations:
-        subscription_in_use_by_ids.setdefault(relation.depends_on_sub_id, []).append(relation.in_use_by_sub_id)
+    if len({key[1] for key in keys}) > 1:
+        raise Exception("Only one inUseByFilter.statuses can be defined")
```

Comment:
Maybe nicer to move this into a reusable function, and use some itertools-fu, so 

```py
def to_subscription_ids_and_filter_statuses(
    keys: list[tuple[UUID, tuple[str, ...] | None]], direction: Literal["dependsOn", "inUseBy"]
) -> tuple[list[UUID], tuple[str, ...]]:
    subscription_ids, filter_statuses_values = unzip(keys)
    filter_statuses = (
        one(
            unique_everseen(filter_statuses_values),
            too_long=Exception(f"{direction}Filter.statuses must be set to the same value"),
        )
        or SubscriptionLifecycle.values()
    )
    return list(subscription_ids), tuple(filter_statuses)
```


Then use like

```py
subscription_ids, filter_statuses = to_subscription_ids_and_filter_statuses(keys, "inUseBy")
```

Then we could have a few isolated unittests for `to_subscription_ids_and_filter_statuses` i.e. for these scenarios


```
Valid example - same status filter set on all load calls: [('subid1', ['active']), ('subid2', ['active']), ('subid3', ['active'])]
subscription_ids=['subid1', 'subid2', 'subid3']
filter_statuses=('active',)

Valid example - no status filter set on any load calls: [('subid1', None), ('subid2', None), ('subid3', None)]
subscription_ids=['subid1', 'subid2', 'subid3']
filter_statuses=('initial', 'active', 'migrating', 'disabled', 'terminated', 'provisioning')

Invalid example - mix of status filter and no status: [('subid1', None), ('subid2', ['active']), ('subid3', None)]
Exception: inUseByFilter.statuses must be set to the same value

Invalid example - mix of status filters: [('subid1', ['active', 'terminated']), ('subid2', ['active']), ('subid3', ['active'])]
Exception: inUseByFilter.statuses must be set to the same value
```